src/main/java/com/kainos/ea/backend/services/JobRolesService.java

```diff
@@ -1,19 +1,63 @@
 package com.kainos.ea.backend.services;
 
+import com.kainos.ea.backend.models.Band;
+import com.kainos.ea.backend.models.Capability;
 import com.kainos.ea.backend.models.JobRole;
 import com.kainos.ea.backend.repositories.JobRolesRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+
 import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 @Service
 public class JobRolesService {
 
+    private JobRolesRepository jobRolesRepository;
+    private BandService bandService;
+    private CapabilityService capabilityService;
+
     @Autowired
-    JobRolesRepository jobRolesRepository;
+    public JobRolesService(JobRolesRepository jobRolesRepository, BandService bandService, CapabilityService capabilityService) {
+        this.jobRolesRepository = jobRolesRepository;
+        this.bandService = bandService;
+        this.capabilityService = capabilityService;
+    }
 
-    public List<JobRole> getAllJobRolesSortedByCapability(){
+    public List<JobRole> getAllJobRolesSortedByCapability() {
         return jobRolesRepository.findAllByOrderByCapability();
     }
 
+    public JobRole saveJobRole(JobRole jobRole) {
+        return jobRolesRepository.save(jobRole);
+    }
+
+    public void addJobRole(JobRole jobRole) throws Exception {
+        Pattern pattern = Pattern.compile("^[A-z][A-z ]+$", Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(jobRole.getName());
+        if (jobRole.getName().length() > 32 || !matcher.find()) {
+            throw new Exception("Invalid role name!");
+        }
+        pattern = Pattern.compile("^[A-z][0-9A-z '().,/-]{0,249}$", Pattern.CASE_INSENSITIVE);
+        matcher = pattern.matcher(jobRole.getName());
+        if (jobRole.getSpecification().length() > 250 || !matcher.find()) {
+            throw new Exception("Invalid specification!");
```

Comment:
Could Exception be too generic? I have used InvalidNameException for a similar thing. Either way, I think we should be consistent

Reply:
Yup, my bad, Michał warned me as well but I forgot to change it. I'll get on it tomorrow and google through the Exceptions to see what would be the most appropriate thing here.